### PR TITLE
Login page: improve UX with form lock and loader

### PR DESCRIPTION
**Pull request by Natsuke**
Hi there,

Small PR to improve the UX on the default Login page.

This include Fields locking and loader on submit Button.

Preview

cap

By the way, great framework ! 👍

### CODE REVIEW

**Comment by fzaninotto** [comment]
Thanks! Could you PR against the `next` branch instead? `master` is only for bugfixes.

Also, I'd love a screenshot of the improved UI in the PR description.

File: src/mui/auth/Login.js [hunk 1]
```diff
@@ -70,7 +71,7 @@ class Login extends Component {
     login = ({ username, password }) => {
         const { authClient, push, location } = this.props;
         if (!authClient) return;
-        authClient(AUTH_LOGIN, { username, password })
+        return authClient(AUTH_LOGIN, { username, password })
```
Comment (fzaninotto) [inline src/mui/auth/Login.js]: Why is it needed?

File: src/mui/auth/Login.js [hunk 2]
```diff
@@ -104,11 +106,19 @@ class Login extends Component {
...
                                         floatingLabelText={translate('aor.auth.password')}
                                         type="password"
+                                        disabled={submitting}
                                     />
                                 </div>
                             </div>
                             <CardActions>
-                                <RaisedButton type="submit" primary disabled={submitting} label={translate('aor.auth.sign_in')} fullWidth />
+                                <RaisedButton
+                                  type="submit"
```
Comment (jpetitcolas) [inline src/mui/auth/Login.js]: Indentation issue

File: src/mui/auth/Login.js (hunk 1 above)
Comment (Natsuke) [inline src/mui/auth/Login.js]: If we follow the `redux-form` doc, the `onSubmit` function must return a promise to handle the `submit` state of the form. This allow usage of the `submitting` props in [Login.js](https://github.com/Natsuke/admin-on-rest/blob/2eff497dfa9c62fb4c86f48af0d74fe747309480/src/mui/auth/Login.js#L80)

cfr: http://redux-form.com/6.5.0/docs/api/ReduxForm.md/#-submit-promise-

File: src/mui/auth/Login.js (hunk 2 above)
Comment (Natsuke) [inline src/mui/auth/Login.js]: Oops ! Editor's config tab size to 2 ! I'll fix it ASAP 😄

File: src/mui/auth/Login.js (hunk 1 above)
Comment (fzaninotto) [inline src/mui/auth/Login.js]: Thanks, makes sense.

**Comment by jpetitcolas** [comment]
Thanks! :)

**Pull request merged**